pkgs/applications/editors/jetbrains/default.nix

```diff
@@ -183,25 +185,34 @@ rec {
       pname = "rider";
       extraBuildInputs = [
         fontconfig
-        stdenv.cc.cc
-        openssl
         libxcrypt
         lttng-ust_2_12
         musl
+        openssl
+        stdenv.cc.cc
       ];
     }).overrideAttrs (attrs: {
-      postInstall = (attrs.postInstall or "") + lib.optionalString (stdenv.isLinux) ''
+      nativeBuildInputs = (attrs.nativeBuildInputs or [ ]) ++ lib.optionals (stdenv.isLinux) [
+        autoPatchelfHook
+      ];
+      buildInputs = (attrs.buildInputs or [ ]) ++ lib.optionals (stdenv.isLinux) [
+        stdenv.cc.cc
+        zlib
+        fontconfig  # plugins/dotTrace/DotFiles/linux-*/libSkiaSharp.so
+      ];
+      dontAutoPatchelf = true;
+      postFixup = (attrs.postFixup or "") + lib.optionalString (stdenv.isLinux) ''
         (
           cd $out/rider
-
-          ls -d $PWD/plugins/cidr-debugger-plugin/bin/lldb/linux/*/lib/python3.8/lib-dynload/* |
```

Comment:
Sorry, I'm just slowly beginning to understand now what's really going on here. I'm afraid I may have mislead you with my initial comment.

The reason `autoPatchelf` is gone is perhaps not because it was mistakenly removed. Actually the `autoPatchelfHook` is already in `bin/linux.nix` and it just unconditionally patches all binaries it finds if `dontAutoPatchelf` is not set. So we don't necessarily need to have the explicit `autoPatchelf` at some point, and we probably also shouldn't remove the line I have commented on here.

As far as I can tell the **actual problem** is simply that the dotnet symlinking doesn't happen properly, because the line removing the original copy was removed. If one looks at the resulting directory in the Nix store then one sees a funny symlink of 

- rider/lib/ReSharperHost/linux-x64/dotnet/ighiv2ngd0w1ccizg95y07kc5zzr5drr-dotnet-sdk-7.0.404
  - to /nix/store/ighiv2ngd0w1ccizg95y07kc5zzr5drr-dotnet-sdk-7.0.404

but it's actually supposed to be

- rider/lib/ReSharperHost/linux-x64/dotnet
  - to /nix/store/ighiv2ngd0w1ccizg95y07kc5zzr5drr-dotnet-sdk-7.0.404

What I think you should do to fix this properly is:

1. Drop the following commits again:
   - [jetbrains.rider: fix dependencies](https://github.com/NixOS/nixpkgs/pull/272842/commits/8e78fc8f58dbb08f5b2bc79f80d9f60c5f2b90ff)
   - [jetbrains.rider: re-introduce autoPatchelf](https://github.com/NixOS/nixpkgs/pull/272842/commits/994803fb989213aeb2cc320c0e5e48cfaf9b41ce)
   - [jetbrains.rider: do not hardcode x64 paths](https://github.com/NixOS/nixpkgs/pull/272842/commits/d3128e692b9542981ea631df7104f183148cf265)
2. Just fix the dotnet symlinking, e.g. with

```diff
diff --git a/pkgs/applications/editors/jetbrains/default.nix b/pkgs/applications/editors/jetbrains/default.nix
index fc73117ec82b..cedde88b3f36 100644
--- a/pkgs/applications/editors/jetbrains/default.nix
+++ b/pkgs/applications/editors/jetbrains/default.nix
@@ -201,6 +201,7 @@ rec {
             --replace-needed libcrypt.so.1 libcrypt.so
 
           for dir in lib/ReSharperHost/linux-*; do
+            rm -r $dir/dotnet
             ln -s ${dotnet-sdk_7} $dir/dotnet
           done
         )
```

I'd keep the `dotnet-sdk` at version 7 because that's what JetBrains normally still ships with Rider.

This makes everything work properly in my quick tests. Sorry for all the back and forth :sweat_smile: